### PR TITLE
Mark uncoverage statements in OpenQA::Worker::WebUIConnection

### PR DESCRIPTION
**Pull request by okurz**
Unreliable coverage observed in tests for
https://github.com/os-autoinst/openQA/pull/3713

Related progress issue: https://progress.opensuse.org/issues/55364

### CODE REVIEW

File: lib/OpenQA/Worker/WebUIConnection.pm [hunk 1]
```diff
@@ -202,15 +202,17 @@ sub _setup_websocket_connection {
...
 
                     # ignore if the connection was disabled from our side
+                    # uncoverable statement
                     return log_debug("Websocket connection to $websocket_url finished from our side.")
                       unless $self->websocket_connection;
 
-                    $reason //= 'no reason';
-                    $self->websocket_connection(undef);
-                    $self->_set_status(
+                    $reason //= 'no reason';               # uncoverable statement
```
Comment (kraih) [inline lib/OpenQA/Worker/WebUIConnection.pm:213]: At least for this one i would just change the `@_` unrolling.
```
my ($self, $code, $reason) = (shift, shift, shift // 'no reason');
```

Comment (okurz) [reply lib/OpenQA/Worker/WebUIConnection.pm]: ok, done in https://github.com/os-autoinst/openQA/pull/3718/commits/bc4dc4aa57ce3f834bd2de1f4edaffa2d44078ce included in this PR

Comment (okurz) [reply lib/OpenQA/Worker/WebUIConnection.pm]: I assume

```perl
finish => sub ($tx, $code, $reason = 'no reason') {
...
```

is not yet part of any stable language definition? ;)

Comment (kraih) [reply lib/OpenQA/Worker/WebUIConnection.pm]: Actually, that's fine too, and works exactly the way you expect it to. 😁

Comment (okurz) [reply lib/OpenQA/Worker/WebUIConnection.pm]: Cool. As tests failed anyway I did that now. Let's see :) Only that my vim syntax checker does not like it yet :D

Comment (kraih) [reply lib/OpenQA/Worker/WebUIConnection.pm]: You have to activate signatures in the namespace (`use Mojo::Base 'Mojo::EventEmitter', -signatures;`). But then it should be fine.

File: lib/OpenQA/Worker/WebUIConnection.pm [hunk 2]
```diff
@@ -194,23 +194,23 @@ sub _setup_websocket_connection {
...
                     # ignore if the connection was disabled from our side
+                    # uncoverable statement
                     return log_debug("Websocket connection to $websocket_url finished from our side.")
                       unless $self->websocket_connection;
 
-                    $reason //= 'no reason';
-                    $self->websocket_connection(undef);
-                    $self->_set_status(
+                    $self->websocket_connection(undef);    # uncoverable statement
+                    $self->_set_status(                    # uncoverable statement
```
Comment (kraih) [inline lib/OpenQA/Worker/WebUIConnection.pm:213]: It seems `websocket_connection` is chainable. So `$self->websocket_connection(undef)->_set_status(` would get rid of another line.

File: lib/OpenQA/Worker/WebUIConnection.pm [hunk 3]
```diff
@@ -194,23 +194,23 @@ sub _setup_websocket_connection {
                     $command_handler->handle_command(@_);
                 });
             $tx->on(
-                finish => sub {
+                finish => sub ($tx, $code, $reason = 'no reason') {
```
Comment (Martchus) [inline lib/OpenQA/Worker/WebUIConnection.pm:197]: You need `use Mojo::Base 'Mojo::EventEmitter', -signatures` to be able to use that syntax here.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: lib/OpenQA/Worker/WebUIConnection.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/Worker/WebUIConnection.pm]: done

File: lib/OpenQA/Worker/WebUIConnection.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Worker/WebUIConnection.pm]: doesn't help vim syntax highlight but should solve the compile problem ;)

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Review by kraih — APPROVED** [review]

**Comment by okurz** [comment]
Anyone has an idea about the codecov related failure:

```
406 Not Acceptable
{'detail': ErrorDetail(string='Could not satisfy the request Accept header.', code='not_acceptable')}
```

?

**Comment by Martchus** [comment]
No, the same happens on #3714 and persisted after re-triggering codecov.

**Comment by kraih** [comment]
Seems to happen for all our PRs now.

**Pull request merged**